app/controllers/shipments_controller.rb

```diff
@@ -0,0 +1,59 @@
+class ShipmentsController < ApplicationController
+respond_to :json
+
+  def estimate
+    begin
+      weight = (package_params[:weight]).to_i
+      dimensions = [(package_params[:length]).to_i,  (package_params[:width]).to_i,  (package_params[:height]).to_i]
+      package = ActiveShipping::Package.new(weight, dimensions)
+      packages = [package]
+      origin = ActiveShipping::Location.new(origin_params)
+      destination = ActiveShipping::Location.new(destination_params)
+      
+      ups = ups_rates(origin, destination, packages)
+      usps = usps_rates(origin, destination, packages)
+      rates = ups.merge(usps)
+      render :json => rates.as_json, :status => :ok
+    rescue
+      render :json => { error: :bad_data, message: "You must provide valid data for package size, package weight, origin location and destination location." }, status: :bad_request
```

Comment:
Most [Ruby style guides](https://github.com/styleguide/ruby) recommend a line length of 80 characters. You can wrap your code by using indentations to show the continuance of the previous line

```render :json => { error: :bad_data, message: "You must provide valid data for package size,"\
                             " package weight, origin location and destination location." }, status: :bad_request